src/sapi.rs

```diff
@@ -259,27 +259,35 @@ impl SpVoice {
 
     fn available_voices(&mut self) -> Vec<Speech::ISpObjectToken> {
         let mut voices = vec![];
+        // Registry keys where voice tokens may be found
+        let voice_categories = [
+            w!(r"HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech\Voices"),
+            w!(r"HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech_OneCore\Voices"),
+        ];
         unsafe {
-            let category: Speech::ISpObjectTokenCategory =
-                syscom::CoCreateInstance(&Speech::SpObjectTokenCategory, None, syscom::CLSCTX_ALL)
-                    .expect("create voice category");
-            category
-                .SetId(
-                    w!(r"HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech\Voices"),
-                    false,
+            for c in voice_categories.iter() {
+                let category: Speech::ISpObjectTokenCategory = syscom::CoCreateInstance(
+                    &Speech::SpObjectTokenCategory,
+                    None,
+                    syscom::CLSCTX_ALL,
                 )
-                .expect("set voice category id");
-
-            let token_enum = category.EnumTokens(w!(""), w!("")).expect("get voice list");
-            loop {
```

Comment:
I wonder if this loop could be cleaned up with [std::iter::from_fn](https://doc.rust-lang.org/std/iter/fn.from_fn.html), what do you think?

Reply:
Huh, that's a good function to know about. This does seem like the appropriate scenario. Is my new change what you had in mind?

Reply:
Yes, that's exactly what I had in mind.